### DISCUSSION TITLE
Differences in usage between Rsbuild and Modern.js builder

### DISCUSSION THREAD

**Discussion by 9aoy** [web-infra-dev/rsbuild#86] [General]
In Rsbuild, some capabilities no longer have built-in support. Such as, TypeScript type checker、babel translation.

**Comment by chenjiahan** [comment]
## Config Changes

- styledComponents config changed: https://github.com/web-infra-dev/rsbuild/issues/24
- `tools.inspector` removed: https://github.com/web-infra-dev/rsbuild/issues/92
- Rename `html.disableHtmlFolder` to `html.outputStructure`: https://github.com/web-infra-dev/rsbuild/issues/14
- `tools.tsChecker` -> `pluginTypeCheck({ forkTsCheckerOptions: {} })` https://github.com/web-infra-dev/rsbuild/issues/65
- `output.disableTsChecker` -> `pluginTypeCheck({ enable: false })` https://github.com/web-infra-dev/rsbuild/issues/65
- `security.checkSyntax` -> `pluginCheckSyntax()` https://github.com/web-infra-dev/rsbuild/issues/263
- `output.polyfill` defaults to `usage` now https://github.com/web-infra-dev/rsbuild/pull/298

## Other Changes

- Default browserslist changed: https://github.com/web-infra-dev/rsbuild/issues/73
- The `modern-web` target is removed: https://github.com/web-infra-dev/rsbuild/issues/64
- plugin name changed: `builderPluginXXX` -> `pluginXXX`
- `builder.serve` is renamed to `rsbuild.preview`
- Most default PostCSS plugins are removed: https://github.com/web-infra-dev/rsbuild/issues/115

**Comment by 9aoy** [comment]
- Not compile JavaScript files under `node_modules`  by default: https://github.com/web-infra-dev/rsbuild/issues/96

### config and plugin changes

- extract `svgr` as a standalone package https://github.com/web-infra-dev/rsbuild/pull/201


### Unsupported configurations:

- output.svgDefaultExport
- output.disableSvgr

**Comment by chenjiahan** [comment]
We can consider creating a `@modern-js/rsbuild-plugin-comat-builder`, and use this plugin to provide the compat configs, this allows PIA to reuse it.

**Comment by 9aoy** [comment]
Use Rsbuild Server instead of Modern.js Server, the api and configurations have some breaking changes.

### related issues
- https://github.com/web-infra-dev/rsbuild/issues/3
- https://github.com/web-infra-dev/rsbuild/issues/444


### configuration changes

- ~~tools.devServer.watch~~
- ~~tools.devServer.liveReload~~
- ~~tools.devServer.before~~  (use dev.setupMiddlewares.unshift instead)
- ~~tools.devServer.after~~  (use dev.setupMiddlewares.push instead)
- ~~tools.devServer.hot~~ (use dev.hmr instead)
- ~~tools.devServer.https~~  (use server.https instead)
- tools.devServer.client => dev.client
- tools.devServer.setupMiddlewares => dev.setupMiddlewares
- tools.devServer.proxy => server.proxy
- tools.devServer.compress => server.compress
- tools.devServer.devMiddleware => dev.writeToDisk
- tools.devServer.headers => server.headers
- tools.devServer.historyApiFallback => server.historyApiFallback
